add Save control

olivierlabayle opened the pull request:
This PR needs:

- [x] merge of https://github.com/JuliaAI/MLJBase.jl/pull/733 and release of MLJBase 0.20

This works locally with https://github.com/JuliaAI/MLJBase.jl/tree/serialization installed.

Todo:

- [x] Update MLJBase dep to 0.20
- [x]  Bump version to 0.5

ablaom commented on src/MLJIteration.jl:
```diff
@@ -39,6 +40,7 @@ include("constructors.jl")
 include("traits.jl")
 include("ic_model.jl")
 include("core.jl")
+include("serialization.jl")
```
This `include` logically needs to precede line 36, or else just dump the code in "controls.jl", which already contains the other controls added by this package. I'd prefer the latter, to be honest.

ablaom commented on src/serialization.jl:
```diff
@@ -0,0 +1,34 @@
+
+Save(filename="machine.jlso"; method=serialize) =
+    Save(filename, method)
+
```
This doesn't quite look right. I want to be able to do `Save(filename="machine.jlso")` but that doesn't work. 

For consistency with existing controls, I think all the following should work:

`Save("machine.jlso")`
`Save("machine.jslo", method=...)`
`Save(filename="machine.jlso")`
`Save(filename="machine.jlso", method=...)`

ablaom commented on src/serialization.jl:
```diff
@@ -0,0 +1,34 @@
+IterationControl.@create_docs(Save,
+             header="Save(filename=\"machine.jlso\")",
+             example="Save(\"run3/machine.jlso\", compression=:gzip)",
+             body="Save the current state of the machine being iterated to "*
```
I see the `header` and `example` are not consistent - one uses `filename` as kwarg, other as regular arg. 

Okay, I guess you've not looked at this doc-strong yet.

ablaom commented on test/serialization.jl:
```diff
@@ -0,0 +1,42 @@
+
+@testset "Save" begin
+    X, y = make_dummy(N=8);
+    @test_throws Exception Save(filename="myfile.jlso")
```
I guess it's confusing to still have .jlso extension here as this is now .jls.

ablaom reviewed (commented):
I guess this is WIP. 

We probably want a test for `method=(filename, mach)->JSLO.save(filename, mach)` or some other format.

olivierlabayle commented:
> I guess this is WIP.

Sorry, my bad I did not review my changes correctly before making the PR. I have taken your comments into consideration:

- Removed the `serialization.jl` file
- Updated docs
- Added a full kw constructor, and parametrized `Save` as for other controls.
- Added a JLSO test.

ablaom reviewed (approved):
Excellent, thanks.
